src/states/node.rs

```diff
@@ -2283,7 +2277,59 @@ impl Node {
             }
             outbox.send_event(Event::Terminate);
         }
-        false
+    }
+
+    // Drop peers to which we think we have a direct or tunnel connection, but where Crust reports
+    // that we're not connected to the peer or tunnel node respectively.
+    fn purge_invalid_rt_entries(&mut self, outbox: &mut EventBox) -> Transition {
```

Comment:
Well, we'd need to pass `self.tunnels` too, so I don't think it's worth it personally.  I can change it though it others agree.